lib/wordpress.js

```diff
@@ -0,0 +1,48 @@
+var _ = require('underscore');
+var params = require('querystring').parse(window.location.search.split('?').pop());
+
+module.exports = {
+  getPostPreviewView: function (view) {
+    // Return false if we don't have everything we need for a preview
+    var isPreviewActive = params._wp_json_nonce && params.is_preview;
+    if (!isPreviewActive) {
+      console.info('Preview is not active.');
+      return false;
+    }
+
+    // Create preview object
+    if (_.isFunction(view)) {
+      view = {template: view};
+    }
+    if (_.isFunction(view.preprocessor)) {
+      var originalPreprocessor = view.preprocessor;
+    }
+    var preview = {
+      resources: {
+        wordpress: {
+          url: 'http://{domain}/wp-json/posts/{post_id}',
+          with_credentials: true,
+          query: { '_wp_json_nonce': params._wp_json_nonce }
+        },
+        revisions: {
+          url: 'http://{domain}/wp-json/posts/{post_id}/revisions',
+          with_credentials: true,
+          query: { '_wp_json_nonce': params._wp_json_nonce }
+        }
+      },
+      preprocessor: function(context) {
+        var lastRevision = context.resources.revisions[0];
+        context.resources.wordpress = _.extend(context.resources.wordpress, lastRevision);
+        if (originalPreprocessor) { context = originalPreprocessor(context); }
+        return context.resources.wordpress;
```

Comment:
This preprocessor is only used for the `wordpress.getPostPreviewView()` function, so it will only be used by WordPress post templates. Those typically [just have the WordPress resource passed in as the context](https://github.com/SparkartGroupInc/jasonaldean.com/blob/master/views/news/%7Bcategory%7D/%7Bid%7D/%7Bslug%7D.hbs#L26). This preprocessor will run other preprocessors first.